src/gui/src/staGui.h

```diff
@@ -125,27 +133,63 @@ struct TimingPathNode
       : pin_(pin),
         is_clock_(is_clock),
         is_rising_(is_rising),
+        is_sink_(is_sink),
         arrival_(arrival),
         required_(required),
         delay_(delay),
         slack_(slack),
         slew_(slew),
-        load_(load)
+        load_(load),
+        sink_node_(nullptr),
+        instance_node_(nullptr)
   {
   }
 
   std::string getNodeName(bool include_master = false) const;
   std::string getNetName() const;
 
+  odb::dbNet* getNet() const;
+  odb::dbInst* getInstance() const;
+  bool hasInstance() const { return getInstance() != nullptr; }
+
+  bool isPinITerm() const { return pin_->getObjectType() == odb::dbObjectType::dbITermObj; }
+  bool isPinBTerm() const { return pin_->getObjectType() == odb::dbObjectType::dbBTermObj; }
+
+  odb::dbObject* getPin() const { return pin_; }
+  odb::dbITerm* getPinAsITerm() const { return static_cast<odb::dbITerm*>(pin_); }
+  odb::dbBTerm* getPinAsBTerm() const { return static_cast<odb::dbBTerm*>(pin_); }
```

Comment:
In methods like these I usually have a check like isPinITerm in getPinAsITerm to avoid any errors propagating and being hard to find after the cast.